Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/PublicBlogController.java

```diff
@@ -0,0 +1,133 @@
+     * generated data for project blog page by fetching project blog entries
+     * from <code>tbl_projectblogentry</code> and checking user privileges.
+     * 
+     * @param projectUnixName
+     *            parameter passed from url as {projectUnixName}
+     * @param model
+     *            model object
+     * @param project
+     *            project instance obtained using @InjectProject annotation
+     * @param principal
+     *            principal object which is required to fetch information about
+     *            logged in user.
+     * @return project blog page containing list of all project blogs from given
+     *         project
+     * @throws QuadrigaStorageException
+     * @throws NoSuchRoleException
+     */
+    @RequestMapping(value = "sites/{ProjectUnixName}/projectblog", method = RequestMethod.GET)
+    public String projectBlog(@PathVariable("ProjectUnixName") String projectUnixName, Model model,
+            @InjectProject(unixNameParameter = "ProjectUnixName") IProject project, Principal principal)
+            throws QuadrigaStorageException, NoSuchRoleException {
+
+        // Fetch blog entries for a project identified by project unix name
+        String projectId = project.getProjectId();
+        Integer count = getBlogEntryCount();
+        List<IProjectBlogEntry> projectBlogEntryList = projectBlogEntryManager.getProjectBlogEntryList(projectId,
+                count);
+
+        // Add the critical data to model object
+        model.addAttribute("projectBlogEntryList", projectBlogEntryList);
+        model.addAttribute("project", project);
+
+        // check the roles of logged in user to decide whether to show add
+        // project link or not. Only project admins and owners are allowed to
+        // add project blogs for a project under consideration
+        if (principal != null) {
+            String userName = principal.getName();
+            if (projectSecurity.isCollaborator(userName, RoleNames.ROLE_PROJ_COLLABORATOR_ADMIN,
+                    project.getProjectId())) {
+                model.addAttribute("isProjectAdmin", true);
+            } else {
+                model.addAttribute("isProjectAdmin", false);
+            }
+
+            if (projectSecurity.isProjectOwner(userName, project.getProjectId())) {
+                model.addAttribute("isProjectOWner", true);
+            } else {
+                model.addAttribute("isProjectOWner", false);
+            }
+        } else {
+            model.addAttribute("isProjectAdmin", false);
+            model.addAttribute("isProjectOWner", false);
+        }
+
+        return "sites/projectblog";
+    }
+
+    /**
+     * reads the number of blog entries to be displayed from user properties
+     * file using {@linkplain Environment} object
+     * 
+     * @return number of blog entries to be displayed specified as
+     *         <code>projectblogentry.list.count<code> property.
+     */
+    private Integer getBlogEntryCount() {
```

Comment:
While developing, I considered that controller should be responsible for specifying the number of entries to fetch since the property is for project blog page only. If we move code to specify the number of entries to manager, manager will always fetch 5 entries specified in properties although some other code might expect the ProjectBlogEntryManager to fetch all the blog entries. That's why I added the specified the number of entries to fetch in controller instead of manager.

Could you please confirm if my understanding is correct? Please let me know if I am missing some case here. I am working on rest of the changes suggested by you.

Thanks,
Pawan